group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -160,11 +161,14 @@ public GroupCoordinatorShard build() {
             if (timer == null)
                 throw new IllegalArgumentException("Timer must be set.");
             if (coordinatorMetrics == null)
-                throw new IllegalArgumentException("CoordinatorMetricsShard must be set.");
+                throw new IllegalArgumentException("CoordinatorMetrics must be set.");
             if (topicPartition == null)
                 throw new IllegalArgumentException("TopicPartition must be set.");
 
             CoordinatorMetricsShard metricsShard = coordinatorMetrics.newMetricsShard(snapshotRegistry, topicPartition);
+            if (!(metricsShard instanceof GroupCoordinatorMetricsShard))
```

Comment:
really minor -- but since the passed in CoordinatorMetrics determines the type of the shard, does it make more sense to check CoordinatorMetrics is of type GroupCoordinatorMetrics and make the error thrown there.

Reply:
The GroupCoordinatorShard is built in CoordinatorRuntime and we don't know which type of CoordinatorMetrics is passed into it. Once we know we're building a GroupCoordinatorShard, only then we can verify the type

Reply:
I was referring to line 162. At that point we can check the metrics passed in and throw the error.

Reply:
ah yes. thanks